File: 016_Selection_Sort/Selection_Sort.cpp
// https://leetcode.com/problems/sort-an-array/description/


#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> sortArray(const std::vector<int>& nums) {
        std::vector<int> sortedNums = nums;

        // Selection sort algorithm
        for (size_t i = 0; i < sortedNums.size() - 1; ++i) {
            size_t minIndex = i;
            for (size_t j = i + 1; j < sortedNums.size(); ++j) {
                if (sortedNums[j] < sortedNums[minIndex]) {
                    minIndex = j;
                }
            }
            std::swap(sortedNums[minIndex], sortedNums[i]);
        }

        return sortedNums;
    }
};

int main() {
    // Example usage
    Solution solution;
    std::vector<int> nums = {64, 25, 12, 22, 11};
    std::vector<int> sortedNums = solution.sortArray(nums);

    // Print sorted array
    for (int num : sortedNums) {
        std::cout << num << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: 016_Selection_Sort/Selection_Sort.py
# https://leetcode.com/problems/sort-an-array/description/


class Solution:
    def sortArray(self, nums):
        sorted_nums = nums.copy()

        # Selection sort algorithm
        for i in range(len(sorted_nums) - 1):
            min_index = i
            for j in range(i + 1, len(sorted_nums)):
                if sorted_nums[j] < sorted_nums[min_index]:
                    min_index = j
            sorted_nums[i], sorted_nums[min_index] = sorted_nums[min_index], sorted_nums[i]

        return sorted_nums

# Example usage
solution = Solution()
nums = [64, 25, 12, 22, 11]
sorted_nums = solution.sort_array(nums)
print(sorted_nums)
